File: BaekJoon/Problems/Step/07/10809.cpp
/* Sample input & output data
baekjoon
1 0 -1 -1 2 -1 -1 -1 -1 4 3 -1 -1 7 5 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
*/


#include <iostream>
#include <string>
#include <vector>

using namespace std;
#define endl "\n"

int main()
{
    // Input
    string s;
    cin >> s;

    // Declare an array for counting a~z
    vector<int> alphabet (26, -1);

    // Count
    for (int i = 0; i < s.length(); i++)
    {
        if (alphabet[s[i] - 'a'] == -1)
        {
            alphabet[s[i] - 'a'] = i;
        }
    }

    // Output
    for (int j = 0; j < 25; j++)    // a ~ y
    {
        cout << alphabet[j] << " ";
    }
    cout << alphabet[25] << endl;   // z

    return 0;
}

File: BaekJoon/Problems/Step/05/10809.go
package main
import "fmt"

func main() {

    // Input
    var s string
    fmt.Scanln(&s)

    // Declare an array for alphabet
    var alphabet []int
    for i := 0; i < 26; i++ {
        alphabet = append(alphabet, -1)
    }
    // fmt.Println(alphabet)                     // test : ok

    // Find the location
    for i := 0; i < len(s); i++ {
        // fmt.Print(s[i] - 'a')                 // test : ok

        if alphabet[s[i] - 'a'] == -1 {
            alphabet[s[i] - 'a'] = i
        }
    }

    // Output
    // fmt.Println(alphabet)                     // test : [……]
    for i := 0; i < 25; i++ {                    // a ~ y
        fmt.Print(alphabet[i], " ")
    }
    fmt.Println(alphabet[25])                    // z
}